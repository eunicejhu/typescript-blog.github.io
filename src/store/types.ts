import { PostsState } from "../features/posts/postsSlice";
import { NotificationsState } from "../features/notifications/notificationsSlice";
import { UsersState } from "../features/users/usersSlice";
export type State = {
  posts: PostsState;
  users: UsersState;
  notifications: NotificationsState;
};
