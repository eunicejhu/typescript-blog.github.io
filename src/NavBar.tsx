import React from "react";
import { Link } from "react-router-dom";

const NavBar = () => (
  <nav>
    <li>
      <Link to="/">Home</Link>
    </li>
    <li>
      <Link to="/notifications">Notifications</Link>
    </li>
  </nav>
);

export default NavBar;
