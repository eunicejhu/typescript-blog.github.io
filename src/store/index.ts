import { configureStore } from "@reduxjs/toolkit";
import postsReducer from "../features/posts/postsSlice";
import usersReducer from "../features/users/usersSlice";
import notificationsReducer from "../features/notifications/notificationsSlice";
import { useDispatch } from "react-redux";

const store = configureStore({
  reducer: {
    posts: postsReducer,
    users: usersReducer,
    notifications: notificationsReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();

export type State = ReturnType<typeof store.getState>;

export default store;
