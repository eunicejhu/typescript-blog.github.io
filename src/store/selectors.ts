// selectors
import { State } from "./index";
import { Notification } from "../features/notifications/notificationsSlice";
import { Post } from "../features/posts/postsSlice";
import { User } from "../features/users/usersSlice";

export const selectAllPosts = (state: State): Post[] => state.posts.data;
export const selectPostById = (
  state: State,
  postId: string
): Post | undefined => state.posts.data.find((post) => post.id === postId);
export const selectAllNotifications = (state: State): Notification[] =>
  state.notifications;
export const selectAllUsers = (state: State): User[] => state.users.data;
