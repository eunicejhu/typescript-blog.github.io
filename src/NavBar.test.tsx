import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import NavBar from "./NavBar";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import store from "./store/index";

const withBrowserRouterAndStore = (ui: React.ReactNode, route = "/initial") => {
  window.history.pushState({}, "", route);
  return (
    <BrowserRouter>
      <Provider store={store}>{ui}</Provider>
    </BrowserRouter>
  );
};
test("click Home, location pathname set to /", () => {
  const ui = withBrowserRouterAndStore(<NavBar />);
  render(ui);
  fireEvent.click(screen.getByText(/Home/i));
  expect(window.location.pathname).toBe("/");
});
test("click Notifications, location pathname set to /notifications", () => {
  const ui = withBrowserRouterAndStore(<NavBar />);
  render(ui);
  fireEvent.click(screen.getByText(/Notifications/i));
  expect(window.location.pathname).toBe("/notifications");
});

test("renders correctly", () => {
  const ui = withBrowserRouterAndStore(<NavBar />);
  const { asFragment } = render(ui);

  expect(asFragment()).toMatchSnapshot();
});
