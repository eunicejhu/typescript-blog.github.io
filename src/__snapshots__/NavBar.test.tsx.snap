// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <nav>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/notifications"
      >
        Notifications
      </a>
    </li>
  </nav>
</DocumentFragment>
`;
